Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -81,35 +85,133 @@ async function getJava(versionSpec: string) {
     toolLib.prependPath(path.join(jdkDirectory, BIN_FOLDER));
 }
 
-function sleepFor(sleepDurationInMillisecondsSeconds): Promise<any> {
-    return new Promise((resolve, reeject) => {
-        setTimeout(resolve, sleepDurationInMillisecondsSeconds);
-    });
+/**
+ * Return file ending if it is supported. Otherwise throw an error.
+ * @param file Path to a file.
+ */
+function getSupportedFileEnding(file: string): string {
+    const fileEnding: string = supportedFileEndings.find(ending => file.endsWith(ending)); 
+
+    if (fileEnding) {
+        return fileEnding;
+    } else {
+        throw new Error(taskLib.loc('UnsupportedFileExtension'));
+    }
+}
+
+/**
+ * Install JDK.
+ * @param sourceFile Path to JDK file.
+ * @param fileExtension JDK file extension.
+ * @param archiveExtractLocation Path to folder to extract a JDK.
+ */
+async function installJDK(sourceFile: string, fileExtension: string, archiveExtractLocation: string, extendedJavaHome: string, versionSpec: string): Promise<string> {
+    let jdkDirectory;
+    if (fileExtension === '.dmg' && os.platform() === 'darwin') {
+        // Using set because 'includes' array method requires tsconfig option "lib": ["ES2017"]
+        const volumes: Set<string> = new Set(fs.readdirSync(VOLUMES_FOLDER));
+
+        await attach(sourceFile);
+    
+        const volumePath: string = getVolumePath(volumes);
+
+        let pkgPath: string = getPackagePath(volumePath);
+        jdkDirectory = await installPkg(pkgPath, extendedJavaHome, versionSpec);
+
+        try {
+            await detach(volumePath);
+        }
+        catch(error) {
+            taskLib.error(error.message);
```

Comment:
Looks like the same issue here

Reply:
Removed try-catch to catch the error on the top level. Thanks!